### ISSUE TITLE
Deploy the agent bundles on Elastic CDN

### ISSUE THREAD

**Issue by jahtalab** [elastic/apm-agent-rum-js#481]
Currently we recommend users to install our package from npm and serve the JavaScript bundle from their own servers. However, providing the RUM agent via our own CDN simplifies testing and deployment for our users. 

I think we should meet the following requirements:
* Ability to publish the final JavaScript bundles to our CDN (this can be done through CI/CD), currently we don't have any CDN endpoints.
* We should have some sort of versioning in the URL, e.g. `https://elastic.co/cdn/apm-rum@4/apm-rum.min.js` for the 4th major version of the agent. It would be better if the version in the url supports semantic versioning ([example](https://unpkg.com/@elastic/apm-rum@4.6.0/dist/bundles/elastic-apm-rum.umd.min.js)) but at the moment we only require supporting the major version (because of breaking changes).
* We should provide high availability, since the JS bundle might be deployed by our users in production and if the endpoint is not available it might result in disrupting their service. This also means that this CDN endpoint could potentially be hit by a huge number of requests.

Depends on:
https://github.com/elastic/apm-agent-rum-js/issues/518

**Comment by jahtalab** [comment]
We're ready to move forward with this task.

cc @v1v ,

**Comment by v1v** [comment]
@olksdr just pointed that the current implementation will show an error message when accessing http://apm-assets.elastic.co/, as long as there is no an index.html file in the root of the bucket. We can deploy a beautiful page with list of the assets (maybe generate it on deployment), what do you think?

**Comment by vigneshshanmugam** [comment]
Yes, We can do that. I am totally in for this. However do we want to showcase only the latest artifacts in the `index.html` or everything that is served through the CDN?

**Comment by jahtalab** [comment]
That's great idea @v1v ! I think we should only show case major versions on that page. We can potentially link to the docs as well.

**Comment by alvarolobato** [comment]
We've discussed this and we are going to move forward, with a phased approach.
- Initially we will host the agent on a CDN with a unique URL per minor version and infinite TTL (we will reevaluate in the future)
- This URLs will be shared with a set of cloud users before making it available to a broader set.
- After this, we will decide if we offer it to all the paid users or to everyone even basic on-prem users.

Not use elastic.co URL, @drewpost was concerned of our domain being identified with a tracker.
cc. @graphaelli

**Comment by jahtalab** [comment]
Reopening since there are multiple phases to complete this task.

**Comment by jahtalab** [comment]
Meeting action item:
- @drewpost to follow up with publishing the elastic CDN to public
